### PR TITLE
CONTRIBUTING.md: Add some style guidelines

### PR DESCRIPTION
**Pull request by akgrant43**
[ci skip]

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -212,3 +212,30 @@ If you were working on a feature branch, do this:
...
+guidelines below, and secondly follow the general rule of keeping to the 
+same formatting conventions as exist in the file.
+
+
+## C Function Declarations
+
+C function declarations should have the type on one line and the 
+function name and parameters on the following line, e.g.:
+
+```
```
Comment (fniephaus) [inline CONTRIBUTING.md]: Adding a *c* to the end of this line should enable C-syntax highlighting.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -212,3 +212,30 @@ If you were working on a feature branch, do this:
...
+convertCopy(char *from, int fromLen, char *to, int toLen, int term)
+```
+
+This facilitates searching for function definitions by searching for the 
+name at the beginning of the line.
+
+
+## Tabs are 4 spaces
+
+The heading says it all.
```
Comment (fniephaus) [inline CONTRIBUTING.md]: I'm afraid the title doesn't answer the question *when to use* spaces/tabs? Are tabs banned?

Comment (nicolas-cellier-aka-nice) [reply CONTRIBUTING.md]: A side note: the legacy platforms/win32 is using these settings:

8-space tabs stops
2-space indent

I don't want to create massive change of ownership for fixing that...
https://stackoverflow.com/questions/4112410/git-change-styling-whitespace-without-changing-ownership-blame

**Comment by akgrant43** [comment]
Hi Fabio & Nicolas,

Thanks for your feedback and my apologies for not responding, I've been caught up with other things the last couple of weeks.  I haven't forgotten and definitely not ignoring this.

Cheers,
Alistair

**Comment by fniephaus** [comment]
No worries at all, there's no rush. Have a great weekend...

Fabio

**Comment by akgrant43** [comment]
There haven't been any more comments for a couple of weeks.  I'm not particularly happy with the section on tabs vs spaces, however it's always easier to criticise someone else's work than start with a blank slate, so take this as inspiration! :-)

[ci skip]

**Pull request merged**